src/zenml/integrations/aws/orchestrators/sagemaker_orchestrator.py

```diff
@@ -523,11 +722,28 @@ def get_pipeline_run_metadata(
         Returns:
             A dictionary of metadata.
         """
-        pipeline_execution_arn = os.environ[ENV_ZENML_SAGEMAKER_RUN_ID]
+        execution_arn = os.environ[ENV_ZENML_SAGEMAKER_RUN_ID]
         run_metadata: Dict[str, "MetadataType"] = {
-            "pipeline_execution_arn": pipeline_execution_arn,
+            "pipeline_execution_arn": execution_arn,
         }
 
+        zenml_client = Client()
+
+        if deployment_id := zenml_client.get_pipeline_run(
+            run_id
+        ).deployment_id:
+            deployment = zenml_client.get_deployment(deployment_id)
```

Comment:
Instead of doing two (hydrated) requests here, you could extend `self.get_settings(...)` to allow inputting a `PipelineConfiguration`, and then fetch that from the step context instead.

Reply:
Not a bad idea. Initially, I was a bit hesitant to change a function that is widely used in other parts of the codebase as well, but adding an additional optional variable shouldn't hurt. I would probably still keep the current code, in case someone wants to call this method outside of a step though. WDYT?

Reply:
As long as you don't remove functionality from it and just add to it, should be good :)
Mhm I guess the current code could be a fallback in case the `get_step_context` call fails? But this method is not supposed to be called outside of a step execution anyway, so I would even take it out to be honest.

Reply:
Fixed this as well.